File: VB/HowToSetValidationRulesInCode.Module/Incident.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation
Imports System.ComponentModel

Namespace HowToSetValidationRulesInCode.Module
	<DefaultClassOptions, DefaultProperty("Subject"), RuleCriteria("RuleCriteria for Incident", DefaultContexts.Save, "AssignedTo is not null", SkipNullOrEmptyValues := False)> _
	Public Class Incident
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private assignedTo_Renamed As Person
		Private subject_Renamed As String
		<RuleRequiredField("RuleRequiredField for Incident.Subject", DefaultContexts.Save)> _
		Public Property Subject() As String
			Get
				Return subject_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Subject", subject_Renamed, value)
			End Set
		End Property
		Public Property AssignedTo() As Person
			Get
				Return assignedTo_Renamed
			End Get
			Set(ByVal value As Person)
				SetPropertyValue("AssignedTo ", assignedTo_Renamed, value)
			End Set
		End Property
	End Class


End Namespace

File: CS/HowToSetValidationRulesInCode.Module/Incident.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using System.ComponentModel;

namespace HowToSetValidationRulesInCode.Module {
	[DefaultClassOptions]
	[DefaultProperty("Subject")]
	[RuleCriteria("RuleCriteria for Incident", DefaultContexts.Save,  "AssignedTo is not null", SkipNullOrEmptyValues = false)]
	public class Incident : BaseObject {
		public Incident(Session session) : base(session) {
		}
		private Person assignedTo;
		private string subject;
		[RuleRequiredField("RuleRequiredField for Incident.Subject", DefaultContexts.Save)]
		public string Subject {
			get {
				return subject;
			}
			set {
				SetPropertyValue("Subject", ref subject, value);
			}
		}
		public Person AssignedTo {
			get {
				return assignedTo;
			}
			set {
				SetPropertyValue("AssignedTo ", ref assignedTo, value);
			}
		}
	}


}
